src/robot/reporting/jsbuildingcontext.py

```diff
@@ -47,6 +48,7 @@ def string(self, string, escape=True, attr=False):
             string = (html_escape if not attr else attribute_escape)(string)
         return self._strings.add(string)
 
+    @functools.lru_cache(maxsize=128)
     def html(self, string):
         return self.string(html_format(string), escape=False)
```

Comment:
I assume time is spent in `html_format`. Another way to avoid calling it with all strings would be just passing the string to `self.string` with an indication that it needs to be HTML formatted. That formatting could then be done when strings are read from `self.strings`.

Reply:
That `self.strings` is kind of a cache already. We just do work before we put stuff there.

Reply:
Do you want to handle this in `self.strings`? I could close the PR if that is a better solution.

Reply:
That was done in 8c08fe7cc47bc2bae3ba51b26c3af0391ff117b1. You could still check does `lru_cache` give additional benefits, but I believe difference is small.

Reply:
Was unmeasurably small. Closing.